### ISSUE TITLE
Crash when RecursionError should be thrown

### ISSUE THREAD

**Issue by OverLordGoldDragon** [ipython/ipython#12387]
```python
class C():
    @property
    def x(self):
        return self.x
C().x
```
See [here](https://stackoverflow.com/q/62356253/10133797); same if running as a file (main) or from within IPython 7.13.0, in Spyder 4.1.3. Some users reproduced this from within terminal (not myself), as attested in comments in linked SO.

**Comment by Carreau** [comment]
Do you know of anyone not on windows who have seen this ?

**Comment by OverLordGoldDragon** [comment]
@Carreau No, anyone on Mac/Linux reported `RecursionError`

**Comment by Carreau** [comment]
So then we need to dig into more details like Python version, installation source and exact os version.

You can _try_ to get more information by manually starting the kernel in one terminal: 

```
$ python -m ipykernel
NOTE: When using the `ipython kernel` entry point, Ctrl-C will not work.

To exit, you will have to explicitly quit this process, by either sending
"quit" from a client, or using Ctrl-\ in UNIX-like environments.

To read more about this, see https://github.com/ipython/ipython/issues/2049


To connect another client to this kernel, use:
    --existing kernel-2311.json
```

And use (for example) In another terminal:

```
$ jupyter console --existing
or
$ jupyter qtconsole --existing
```
(then enter failing example)

With that if there is any crash with error messages going nowhere you have more chance of seeing them.

trying to lower or increase recursion limit may help find the problem: 

```
sys.setrecursionlimit(...)
```

On my machine I can get IPython to "work" with a recursion limit between ~120 and ~19000 below or above that there are issues.

Other question is whether this is reproducible in pure terminal IPython and not ipykernel. If it look "stuck" I'm hoping it's not a race condition, or a file descriptor that needs to be flushed. I'm not too sure how to track that on windows.

**Comment by ccordoba12** [comment]
You're right @Carreau! I changed the recursion limit to 120 and IPython didn't crashed!

The problem seems that IPython tries to print a lot of tracebacks for this:

![imagen](https://user-images.githubusercontent.com/365293/84621104-53f7c000-ae9f-11ea-9913-48446cda93ae.png)

**Comment by ccordoba12** [comment]
> So then we need to dig into more details like Python version, installation source and exact os version.

I can provide that info:

* Python 3,7
* Conda installation
* Windows 7

**Comment by Carreau** [comment]
Interesting, .. .so the code that detect recursions can't for whatever reason...

Be careful if you poke at the code, in IPython master we rely on `stack_data` (a new package) and not on 7.x where I believe the detection is in `ultratb.py`. I don't have a windows machine or VM setup yet, so if you figure it out great. Otherwise it may take me some time to get to it.

**Comment by OverLordGoldDragon** [comment]
@Carreau What I get with `jupyter qtconsole --existing`: 

![image](https://user-images.githubusercontent.com/16495490/84629832-03fb1680-aefc-11ea-96a1-f241e5a42e25.png)

For me likewise, `sys.setrecursionlimit(120)` resolves it within Spyder's IPython. 121 also worked, so I figured I'd find the tipping point: **2688-2691**. It's a range not because I didn't test, but because _each of these_ (except 2691) works _sometimes_. Python 3.7.7.

**Comment by Carreau** [comment]
It feel to me like a deadlock where the size of the frame to print fill the iopub buffer on one side, and the buffer is never read from the other.
